rust/Dockerfile

```diff
@@ -24,47 +10,15 @@ RUN set -xe \
   ca-certificates \
   gcc
 
-ENV RUSTUP_HOME=/usr/local/rustup \
-  CARGO_HOME=/usr/local/cargo \
-  PATH=/usr/local/cargo/bin:$PATH
-
-ARG RUSTUP_VERSION
-ARG RUSTUP_x86_DOWNLOAD_SHA256
-ARG RUSTUP_aarch64_DOWNLOAD_SHA256
-ARG RUST_VERSION
-RUN set -eux; \
-  apkArch="$(apk --print-arch)"; \
-  case "$apkArch" in \
-  x86_64) rustArch='x86_64-unknown-linux-musl'; rustupSha256=${RUSTUP_x86_DOWNLOAD_SHA256} ;; \
-  aarch64) rustArch='aarch64-unknown-linux-musl'; rustupSha256=${RUSTUP_aarch64_DOWNLOAD_SHA256} ;; \
-  *) echo >&2 "unsupported architecture: $apkArch"; exit 1 ;; \
-  esac; \
-  url="https://static.rust-lang.org/rustup/archive/${RUSTUP_VERSION}/${rustArch}/rustup-init"; \
-  wget "$url"; \
-  echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
-  chmod +x rustup-init; \
-  ./rustup-init -y --no-modify-path --profile minimal --default-toolchain ${RUST_VERSION} --default-host ${rustArch}; \
-  rm rustup-init; \
-  chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
-  rustup --version; \
-  cargo --version; \
-  rustc --version;
-
 # This image is used to prepare Cargo Chef which is used to cache dependencies
 FROM rust as chef
-
-ARG CARGO_CHEF_VERSION
 RUN set -xe \
-  && apk add --no-cache musl-dev  \
-  && cargo install cargo-chef --locked --version=${CARGO_CHEF_VERSION} \
```

Comment:
Skipping `cargo-chef` takes 3 minutes off the wall time for CI, but I'm in violation of Chesterton's Fence here, I don't know what `cargo-chef` is for. Does it speed up local dev builds of the images?